proto/poktroll/application/application.proto

```diff
@@ -15,4 +15,23 @@ message Application {
   cosmos.base.v1beta1.Coin stake = 2; // The total amount of uPOKT the application has staked
   repeated poktroll.shared.ApplicationServiceConfig service_configs = 3; // The list of services this appliccation is configured to request service for
   repeated string delegatee_gateway_addresses = 4 [(cosmos_proto.scalar) = "cosmos.AddressString", (gogoproto.nullable) = false]; // The Bech32 encoded addresses for all delegatee Gateways, in a non-nullable slice
+  repeated ArchivedDelegations archived_delegations = 5 [(gogoproto.nullable) = false];
 }
+
+// ArchivedDelegations defines the type used to store the archived delegations
+// for an application in addition to the last block height where the delegation
+// was active.
+message ArchivedDelegations {
```

Comment:
Is there a reason we went with a list `[(height, [add])]` rather than map `map[height][[addr]]`?

Reply:
Had the misconception that protobufs do not support maps. Will see how this simplifies the implementation :+1:

Reply:
Sorry for not making that clear. Whenever I said "map" in our conversations. I meant it LITERALLY.

I genuinely think we (gut intuition) we can potentially get rid of half of of the business logic in this PR and avoid any new storage functions.

Before you reimplement the whole thing (and especially before you write unit / integration tests), let's make sure to align if you're unsure.

![Screenshot 2024-05-02 at 6 59 23 PM](https://github.com/pokt-network/poktroll/assets/1892194/d08206c7-2c82-435a-afb1-064f41022bc3)